### ISSUE TITLE
Supporting `sentence-transformers-3.x.x`

### ISSUE THREAD

**Issue by yassinehamdouni** [superlinked/superlinked#54]
**Is your feature request related to a problem? Please describe.**
Was trying to upgrade from `sentence-transformer-2.2.2` (the default that `superlinked-6.4.0` comes with) to `sentence-transformers-3.0.1`, to test out the new embedding models.  

However, getting this error:
`ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
superlinked 6.4.0 requires sentence-transformers==2.2.2, but you have sentence-transformers 3.0.1 which is incompatible.`

Do y'all plan on making upcoming versions of `superlinked` compatable with `sentence-transformers-3.x.x` ?

**Comment by ClaireSuperlinked** [comment]
Thanks for flagging @yassinehamdouni - we will take a look at this. @kembala are you able to support?

**Comment by kembala** [comment]
We have had a brief look into this and concluded that using 3.x is significantly slower on our end. We have yet to invest into a discovery whether that is due to some different configuration needs and have opted to not pursue this upgrade just yet. Let me know, what you see on your end.

**Comment by yassinehamdouni** [comment]
Gotcha! Do you have a ball-park timeline for when you plan on upgrading?

**Comment by kembala** [comment]
> Gotcha, no worries! Do you have a ball-park timeline for when you plan on upgrading?

In general our approach is to run experiments first and productize changes that are proven to lift the quality of the framework. If you have such observations at hand, we are always open for a discussion. Paging @morkapronczay who is spearheading such efforts on our end.

TLDR: No timeline on this change yet, we are focusing our efforts towards a server implementation that let's you leverage the current SL ecosystem with different VDBs.

**Comment by yassinehamdouni** [comment]
@kembala @morkapronczay any chance we can get an update to `sentence-transformers-2.5.1` ? there's some high-performing models, such as `mixedbread-ai/mxbai-embed-large-v1` (https://huggingface.co/mixedbread-ai/mxbai-embed-large-v1), that are only available starting in `sentence-transformers-2.5.1`

**Comment by kembala** [comment]
Superlinked now works with `sentence-transformers = "3.0.1"`. 

We have found out that sentence-transformers had a change in it's caching logic and we have applied some patching on top to make it performant in our context. Looking forward to seeing what you are cooking up with the new models 🤞
